### PR TITLE
enhance: Support memory mode chunk cache (#35347)

### PR DESCRIPTION
**Pull request by bigsheeper**
Chunk cache supports loading raw vectors into memory.

issue: https://github.com/milvus-io/milvus/issues/35273

pr: https://github.com/milvus-io/milvus/pull/35347

### CODE REVIEW

**Comment by czs007** [comment]
e2e is not triggered

**Comment by bigsheeper** [comment]
need to pick https://github.com/milvus-io/milvus/pull/36276 after this pr merged

File: internal/core/src/segcore/SegmentSealedImpl.cpp [hunk 1]
```diff
@@ -893,7 +897,10 @@ std::tuple<std::string, std::shared_ptr<ColumnBase>> static ReadFromChunkCache(
     const storage::ChunkCachePtr& cc,
     const std::string& data_path,
     const storage::MmapChunkDescriptorPtr& descriptor) {
-    auto column = cc->Read(data_path, descriptor);
+    // For mmap mode, field_meta is unused, so just construct a fake field meta.
+    auto fm = FieldMeta(FieldName(""), FieldId(0), milvus::DataType::NONE);
```
Comment (czs007) [inline internal/core/src/segcore/SegmentSealedImpl.cpp:901]: Is there a way to specify the right data type even though it is not need for now ?

**Pull request merged**